### PR TITLE
cache calls to prison API

### PR DESCRIPTION
**Pull request by whitfield-mj**

### CODE REVIEW

File: server/services/userService.ts [hunk 1]
```diff
@@ -30,7 +33,28 @@ export default class UserService {
...
+  async getUserCaseLoads(token: string, username: string): Promise<CaseLoad[]> {
+    try {
+      await this.ensureConnected()
+      const redisData = await this.redisClient.get(username)
+      if (redisData) {
+        return JSON.parse(redisData).caseLoads as CaseLoad[]
+      }
+
+      const caseLoads = await this.prisonApiClientBuilder(token).getUserCaseLoads()
+      if (caseLoads.length <= 9) {
```
Comment (petergphillips) [inline server/services/userService.ts:51]: don't really understand why it is a `9` here - are you relying on each caseload response having at least 5 entries?  Couldn't you use the size of the array instead?

File: server/services/userService.ts [hunk 2]
```diff
@@ -30,7 +33,28 @@ export default class UserService {
...
+        await this.redisClient.set(username, JSON.stringify({ caseLoads }), { EX: 300 })
```
Comment (petergphillips) [inline server/services/userService.ts:52]: what does `EX: 300` mean?

File: server/services/userService.ts (hunk 1 above)
Comment (whitfield-mj) [reply server/services/userService.ts:51]: Ha, good spot! That was for testing as my user has more than 1

Comment (whitfield-mj) [reply server/services/userService.ts:51]: As caught by my int test

File: server/services/userService.ts (hunk 2 above)
Comment (whitfield-mj) [reply server/services/userService.ts:52]: It expires in 300s

**Review by davidmiddletonmoj — APPROVED** [review]

**Review by danielburnley — APPROVED** [review]

**Pull request merged**